classes/ObjectModel.php

```diff
@@ -1876,15 +1876,28 @@ public function deleteImage($force_delete = false)
      *
      * @return bool
      */
-    public static function existsInDatabase($id_entity, $table)
+    public static function existsInDatabase($id_entity, $table = null)
     {
+        if ($table !== null) {
+            $primary = 'id_' . bqSQL($table);
+        } else {
+            $object_def = static::$definition;
+
+            if (!isset($object_ref['table']) || !isset($object_ref['primary'])) {
+                return false;
+            }
+
+            $table = $object_def['table'];
+            $primary = $object_def['primary'];
+        }
+
         $row = Db::getInstance()->getRow(
-            '
-			SELECT `id_' . bqSQL($table) . '` as id
-			FROM `' . _DB_PREFIX_ . bqSQL($table) . '` e
-			WHERE e.`id_' . bqSQL($table) . '` = ' . (int) $id_entity,
-            false
-        );
+                (new DbQuery())
+                    ->select('e.`' . $primary . '` as id')
+                    ->from($table, 'e')
+                    ->where('e.`' . $primary . '` = ' . (int) $id_entity),
+                true
```

Comment:
```suggestion
                false
```

Reply:
@PierreRambaud  Why are you disabling the cache? 🤔

Reply:
I agree with @PierreRambaud using cache is often prone to mistake (especially in PrestaShop where it's used badly quite often ^^)
For a critical method like this one that checks if an object exists we shouldn't cache the result If the object has been deleted during the process, the method will keep returning that it exists And we have no easy way to clear this cache on deletion either so we can't keep it consistent

Reply:
I agree as well. I didn't pay attention to this point in my previous suggestion.

Reply:
Thanks guys !

Reply:
PR from @PululuK #21616